client/index.html

```diff
@@ -1,3 +1,48 @@
 <!DOCTYPE html>
-<main></main>
-<script src="./index.jsx" type="module"></script>
+<html>
+  <head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+
+    <link
+      rel="stylesheet"
+      type="text/css"
+      href="https://cdn.jsdelivr.net/npm/bootstrap@5.1.1/dist/css/bootstrap-reboot.css"
+    />
+    <link rel="stylesheet" type="text/css" href="./main.css" />
+  </head>
+  <body>
+    <main class="layout">
+      <section class="wrapper">
+        <article class="content">
+          <p>
+            Today, we’re launching the <em>New York</em> Magazine cover store,
+            where you can buy some of the iconic covers that the magazine has
+            created over its 53-year history. We’re starting with about 50
+            covers from our archives, from our very first issue from April 1968
+            to this year’s Fall Preview showcasing the cast of
+            <em>Succession</em>.
+          </p>
+          <div class="product-card"></div>
+          <div class="product-card"></div>
+          <p>
+            There are some other recent favorites, like the New Yawk merch
+            monster from our Spring Fashion issue and our “Reasons to Love New
+            York” cover where we paid tribute to all the places that closed
+            since the pandemic began. You’ll find some classic Milton Glaser
+            illustrations, including “A Gentile’s Guide to Jewish Food” from
+            July 1968 and “Why Everybody’s Talking About Gossip” from May 1976.
+          </p>
+          <div class="product-card"></div>
+          <p>
+            We’ve also included some covers and designs you’ve never been able
+            to purchase before, like the one where Bert Stern recreated Marilyn
+            Monroe’s last (nude) photo shoot with Lindsay Lohan.
+          </p>
+        </article>
+      </section>
+      <aside class="right-rule"></aside>
+    </main>
+    <script type="module" src="./main.js"></script>
+  </body>
+</html>
```

Comment:
This might be something to discuss voice-to-voice—I don’t have a strong opinion, but I do have a long opinion. The TL;DR for what I was typing out is, JSX is totally baffling to folks who haven’t worked in React, and React class components are considered obsolete to someone who has. Nobody wins.

My assumption and expectation is that a mid- or senior-level candidate _should_ have a preference—even if that’s just HTML, CSS, and jQuery—and should know how their preferred tool it binds a component to the DOM. (Even knowing or not knowing that detail feels like a stronger signal than picking up JSX/classes on the fly.)

We could pick a tool, or have boilerplate bootstrap code ready to hand for several tools, or stick to vanilla JS. The hope is that fleshing out the starting HTML will compensate for the time lost in booting the app. May be something to figure out when we do dry runs.

> _productive_
![](https://garb.ag/martharoll.gif)

Reply:
haha, perhaps that's the truth of the world, built on long opinions, as opposed to strong opinions.

I feel you on the jsx critique, and agree that it's very fair to ask the candidate to express opinions on their tooling of choice. But I think that line of questioning isn't necessarily exclusive to the technical challenge, and we could, alternatively, query that in the team/peer interview instead.

I think we mentioned in the past the benefit of adding a `vue` branch to the existing challenge, and I think that would be a good option going forward, to have equivalent boilerplate in just Vue and React at the beginning of the challenge.

This may be a fairly minor point ultimately. Big picture, I appreciate this iteration on the pairing challenge!

One other thought that's occurred to me: in the past we've used an exercise in which the candidate is asked to review a Pull Request. I'm curious to get thoughts on whether we might reproduce some aspect of that informally in this exercise: to scaffold out some larger piece of work and have the candidate assess it? I think this would also help to establish a point of baseline comparison across candidates

Reply:
I had a similar thought—just dropping in whatever binding code is necessary for a handful of popular libraries. The place I wanted to get to with the starter code was a not-very-good but functioning blob of markup and JS, on the theory that you could more or less copy and paste the boilerplate into any familiar frontend SPA as a starting point. That plus a commented-out `ReactDOM.render` alongside whatever Vue uses might be enough.

The important thing, to my mind, is providing a stronger baseline, in order to give the candidate more space to improvise. The interviews that provided the strongest signals were those when the candidate moved quickly enough through the predefined changes that we could work a little on the `TubeItem` child component’s boundaries and contracts, or on which CSS layout patterns made the most sense for an unordered list with complex child semantics. Optimizing the pairing challenge to get us to that stage faster will get us a much clearer read on candidates going forward.

I did the PR review when I applied and thought it was great. Particularly for senior candidates: it’s a good litmus test for how well they can balance rigor and respect. As I recall, it was in lieu of a code challenge. I’m not sure I could find 45 minutes of conversation in there, but I would certainly be open to helping slot it into the process.

Reply:
I think that's really well put, and I think having the right balance of pre-defined challenge and allowing more advanced candidates to express product thinking through a more open-ended framework will improve on the current strategy while still allowing us to assess both junior and senior candidates without retooling the challenge